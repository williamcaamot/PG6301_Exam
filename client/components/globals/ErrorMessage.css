.errorMessage {
    background-color: #ffdddd;
    border-left: 6px solid #f44336;
    padding: 12px;
    color: #d8000c;
    border-radius: 4px;
    font-family: Arial, sans-serif;
    font-size: 16px;
    line-height: 24px;
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}