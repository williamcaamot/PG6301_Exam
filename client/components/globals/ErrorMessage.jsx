import React from "react";

function ErrorMessage(props) {
  if (props.message) {
    return (
      <>
        <div
          style={{ width: "100%", paddingTop: "10px", paddingBottom: "10px" }}
        >
          <div className={"errorMessage"}>
            <p>{props.message}</p>
          </div>
        </div>
      </>
    );
  }
}

export default ErrorMessage;
