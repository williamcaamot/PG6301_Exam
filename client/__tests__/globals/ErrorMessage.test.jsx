import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import ErrorMessage from "../../components/globals/ErrorMessage.jsx";

describe("error message", () => {
  it("show a error message", () => {
    const message = "This is an error!";

    const component = renderer.create(
      <MemoryRouter>
        <ErrorMessage message={message} />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
