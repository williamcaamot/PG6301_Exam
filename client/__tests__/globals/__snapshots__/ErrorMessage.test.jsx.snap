// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error message show a error message 1`] = `
<div
  style={
    {
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "width": "100%",
    }
  }
>
  <div
    className="errorMessage"
  >
    <p>
      This is an error!
    </p>
  </div>
</div>
`;
